docs/pages/guides/schema-validation.mdx

```diff
@@ -0,0 +1,124 @@
+
+In the case where schema validation is enforced, when the client performs an operation that updates the Storage Tree, it is sent to the Liveblocks server which verifies that the updated Storage Tree is consistent with the defined schema attached to the Room.  If the verification succeeds, the Op is acknowledged by the initial client, and the Storage Tree is updated and forwarded to other users (clients) in the room.  If the operation verification fails, a `RejectedOpsServerMsg` is sent to the client.  The Storage Tree is not updated. In this instance, it is the responsibility of the developer to resolve the error.
+
+## Create a schema
+To create a schema, you will need to POST to the schemas endpoint:
+```ts
+POST https://api.liveblocks.io/v2/schemas
+{
+  "id": "some-id",
+  "name": "Your schema name",
+  "data": `{
+    // your data schema will go here
+  }`,
+}
+
+type Body = {
+  id: string, // MAX_LENGTH = 64 / allowed characters az-
+  name: string, // MAX_LENGTH = 256
+  data: string,
+}
+```
+Note that the data type will be of type string because the schema will be string interpolated. 
+The operation above creates the schema, but it has not yet been assigned to a Room.  To attach this schema to a Room, you will POST to the rooms endpoint:
+```ts
+POST https://api.liveblocks.io/v2/rooms/{id}/schema
+{
+  "schema": "some-id"
+}
+```
+## The validation process in practice
+If you would like to work through an example, you can follow along by using the Multiplayer Form example.
+
+The schema we will define for the example is the following:
+```ts
+POST https://api.liveblocks.io/v2/schemas
+
+{
+  "id": "collaborative-form",
+  "name": "Collaborative Form",
+  "data": `
+    type Logo {
+      name: String
+      theme: String
+    }
+    type Storage {
+      logo: LiveObject<Logo>
+    }
+  `
+}
+// 200 Response
+{
+   "id": "collaborative-form",
+   "name": "Collaborative Form",
+   "data": "schema",
+   "version": 1,
+   "createdAt": "2023-01-22T15:10:25.225Z",
+   "updatedAt": "2023-01-22T15:10:25.225Z"
+}
+```
+
+To attach this schema to a Room, you will POST to the rooms endpoint:
+
+```ts
+POST https://api.liveblocks.io/v2/rooms/{id}/schema
+{
+  "schema": "collaborative-form@1"
+}
+```
+
+To simulate the importance of schema validation and also demonstrate how operations can fail, we will change nested value of `name` to `organization`:
+
+```tsx file="_app.tsx"
+    <RoomProvider
+      id={roomId}
+      initialPresence={{ focusedId: null }}
+      initialStorage={{
+        logo: new LiveObject({
+          organization: "Acme Inc.",
+          theme: "light",
+        }),
+      }}
+    >
+```
```

Comment:
Great example, but it won't work in `initialStorage` unfortunately (because that will only populate an _empty_ room). If you want to illustrate this failure, you will have to make the change in the "onChange" handler in this example, where it runs the `root.get('logo').set('name', e.target.Value)` code and change it to `root.get('logo').set('organisation', e.target.Value)` instead.

While at it, you can also illustrate other/different failures:
- `root.get('logo').set('name', 42)` (not a string value)
- `root.get('logo').delete('name')` (cannot delete the "name" field as it isn't optional)

Reply:
We're working on those details still, but early next week we should have a better picture of it in detail. Trying to get it all worked out before the launch.

Reply:
perfect thanks, just wanted to make sure i wasn't missing something here :)